packages/validation/src/type-checker.js

```diff
@@ -132,7 +133,7 @@ function checkStacks(expectedStack, actualStack) {
     }
 
     // There are still types left on the resulting stack
-    if (j >= 0) {
+    if (actualStack.slice(0, j + 1).filter(t => t !== POLYMORPHIC).length > 0) {
```

Comment:
Could we extract this into a new function? It's difficult to understand to me

Reply:
Sure, I'll make it more explicit. This is saying "If the stack contains anything else than polymorphic types". The reason is that a polymorphic type can also be `[]` so its valid in that case even though there are still types left on the stack after the function.